inference/core/models/roboflow.py

```diff
@@ -294,92 +206,112 @@ def get_model_artifacts(self) -> None:
-                    success = False
-                    attempts = 0
-                    while not success and attempts < NUM_S3_RETRY:
-                        try:
-                            s3.download_file(
-                                INFER_BUCKET,
-                                f"{self.endpoint}/{f}",
-                                self.cache_file(f),
-                            )
-                            success = True
-                        except Exception as e:
-                            attempts += 1
-                            logger.error(
-                                f"Failed to download model artifacts after {attempts} attempts | Infer Bucket = {INFER_BUCKET} | Object Path = {self.endpoint}/{f} | Weights File = {self.weights_file}",
-                                traceback.format_exc(),
-                            )
-                            sleep(SLEEP_SECONDS_BETWEEN_RETRIES)
-                    if not success:
-                        raise Exception(f"Failed to download model artifacts.")
-
-                self.load_environment_artifacts_from_cache()
-            else:
-                self.log("Downloading model artifacts from Roboflow API")
-                # AWS Keys are not available so we use the API Key to hit the Roboflow API which returns a signed link for downloading model artifacts
-                self.api_url = ApiUrl(
-                    f"{API_BASE_URL}/ort/{self.endpoint}?api_key={self.api_key}&device={self.device_id}&nocache=true&dynamic=true"
-                )
-                api_data = get_api_data(self.api_url)
-                if "ort" not in api_data.keys():
-                    raise TensorrtRoboflowAPIError(
-                        f"An error occurred when calling the Roboflow API to acquire the model artifacts. The endpoint to debug is {self.api_url}. The error was: Key 'tensorrt' not in api_data."
-                    )
-
-                api_data = api_data["ort"]
+        infer_bucket_files.append(self.weights_file)
+        logger.info(f"List of files required to load model: {infer_bucket_files}")
+        return infer_bucket_files
+
+    def download_model_artefacts_from_s3(self) -> None:
+        try:
+            logger.debug("Downloading model artifacts from S3")
+            infer_bucket_files = self.get_all_required_infer_bucket_file()
+            cache_directory = get_cache_dir(model_id=self.endpoint)
+            s3_keys = [f"{self.endpoint}/{file}" for file in infer_bucket_files]
+            download_s3_files_to_directory(
+                bucket=INFER_BUCKET,
+                keys=s3_keys,
+                target_dir=cache_directory,
+                s3_client=S3_CLIENT,
+            )
+        except Exception as error:
+            raise ModelArtefactError(
+                f"Could not obtain model artefacts from S3. Cause: {error}"
+            ) from error
+
+    def download_model_artefacts_from_roboflow_api(self) -> None:
```

Comment:
Can we use "artifact" instead of "artefact" ? :-)

Reply:
👍